Examples/HDL/Digital_Filtering/HDL_DUC/README.md

```diff
@@ -0,0 +1,93 @@
+
+See also the [DUC implemented on AI Engine devices](https://github.com/Xilinx/Vitis_Model_Composer/tree/2024.2/Examples/AIENGINE/DSPlib/DUC).
+
+## Algorithm
+
+The DUC design consists of multi-stage finite impulse rate (FIR) filters, a direct digital synthesizer (DDS) and a mixer. 
+
+![](./Images/duc_block_diagram.png) 
+
+The DUC specification is as follows:
+* 4 stages of FIR filter, with interpolation ratio of 2 in each stage and an overall interpolation ratio of 16.
+* The first stage is a 64 tap square raised root cosine (SRRC) filter, and the next three stage filters are half-band (HB) interpolate by 2 FIR filters. 
+
+## Example Model
+
+The example model compares a Simulink implementation of the DUC with an HDL implementation.
+
+![](./Images/model.png) 
+
+The input to both designs is white noise with a sample rate of 50 MSPS.
+
+![](./Images/input_signal.png) 
+
+### Simulink Reference Design
+
+In the reference design, the 4 filter stages are implemented with Simulink's **FIR Interpolation** block.
+
+![](./Images/simulink_reference.png) 
+
+The **Gain** blocks between each stage control bit growth and ensure that the input to each filter is 32 bits.
+
+### PL DUC Design
+
+The filters in the PL design are implemented using the FIR Compiler block from the **HDL/DSP/AXI-S** library.
+
+![](./Images/pl_model.png) 
+
+The **Shift** blocks between each stage control bit growth and ensure that the input to each filter is 17 bits.
+
+The mixer is implemented using:
+
+* DDS Compiler block to implement a complex sinusoid.
+* DSP58 blocks to implement a complex multiply operation. 4 DSP58s are arranged in cascade to achieve high performance and close timing at 800 MHz.
+
+![](./Images/mixer.png) 
+
+![](./Images/cascade.png) 
+
+## Results
+
+### DUC Output
+
+The DUC upconverts the white noise input to center it on 80 MHz with a bandwidth of 50 MHz. The sample rate of the output signal is 800 MSPS. The PL implementation is compared to the Simulink golden reference model. Note the raised noise floor of the PL implementation compared to the Simulink floating-point golden reference.
+
+![](./Images/duc_output.png) 
+
+### Hardware Implementation
+
+In the **Vitis Model Composer Hub** block, the selected hardware is a Versal VCK190 evaluation platform. This board contains a Versal AI Core `xcvc1902-vsva2197-2MP-e-S` part with `-2MP` speed grade.
+
+![](./Images/hub2.png) 
+
+Also note on the **Code Generation->Settings** tab that the design is configured for an FPGA clock period of 1.25 ns (rate of 800 MHz). This clock constraint will be passed to Vivado for implementation.
+
+![](./Images/hub1.png) 
+
+### Timing Closure
+
+To determine whether or not the design will meet timing, run Timing Analysis from the **Analyze** tab. After running Vivado Synthesis and Implementation, the **Timing Analyzer** window will display the post implementation critical paths of the design.
```

Comment:
It would be good to have a screenshot to show the Timeing Closures and and Analyze button. Right now it is not clear where these thigns are.